website_sale_product_minimal_price/models/product_template.py

```diff
@@ -12,12 +14,31 @@ class ProductTemplate(models.Model):
     )
 
     def _compute_has_distinct_variant_price(self):
+        partner = self.env.context.get('partner')
+        pricelist_id = self.env.context.get('pricelist')
         for template in self:
             if template.product_variant_count > 1:
-                prices = template.product_variant_ids.mapped('website_price')
-                if len(prices) > 1:
+                prices = template._get_variants_price(pricelist_id, 1, partner)
+                if len(set(prices.values())) > 1:
                     template.has_distinct_variant_price = True
 
+    # TODO: Use ormcache but we need invalidate cache in all models involved
+    #  in a price change
```

Comment:
I don't recommend merging this as it is.

When using cache, AFAIK you just cannot leave a TODO like this! It has big probabilities to produce bugs...